cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java

```diff
@@ -17,87 +17,90 @@
 package co.cask.cdap;
 
 import co.cask.cdap.api.app.AbstractApplication;
-import co.cask.cdap.api.mapreduce.AbstractMapReduce;
-import co.cask.cdap.api.mapreduce.MapReduceContext;
 import co.cask.cdap.api.workflow.AbstractWorkflow;
-import co.cask.cdap.internal.app.runtime.batch.WordCount;
-import org.apache.hadoop.mapreduce.Job;
+import co.cask.cdap.api.workflow.AbstractWorkflowAction;
+import co.cask.cdap.api.workflow.WorkflowContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.Map;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
 
 /**
  *
  */
 public class WorkflowAppWithFork extends AbstractApplication {
+  public static final String SYNCH_ON_FILE = "WorkflowAppWithFork.done";
   @Override
   public void configure() {
     setName("WorkflowAppWithFork");
     setDescription("Workflow App containing fork");
-    addMapReduce(new OneWordCountMapReduce());
-    addMapReduce(new AnotherWordCountMapReduce());
     addWorkflow(new WorkflowWithFork());
   }
 
   /**
    *
    */
-  public static final class OneWordCountMapReduce extends AbstractMapReduce {
+  public static class WorkflowWithFork extends AbstractWorkflow {
 
     @Override
     public void configure() {
-      setName("OneWordCountMapReduce");
-      setDescription("WordCount job from Hadoop examples");
-    }
-
-    @Override
-    public void beforeSubmit(MapReduceContext context) throws Exception {
-      Map<String, String> args = context.getRuntimeArguments();
-      String inputPath = args.get("oneInputPath");
-      String outputPath = args.get("oneOutputPath");
-      WordCount.configureJob((Job) context.getHadoopJob(), inputPath, outputPath);
-    }
-
-    @Override
-    public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
-      // No-op
+      setName("WorkflowWithFork");
+      setDescription("WorkflowWithFork description");
+      fork().addAction(new OneAction()).also().addAction(new AnotherAction()).join();
     }
   }
 
   /**
    *
    */
-  public static final class AnotherWordCountMapReduce extends AbstractMapReduce {
+  static class OneAction extends AbstractWorkflowAction {
+    private static final Logger LOG = LoggerFactory.getLogger(OneAction.class);
+    private String doneDir = "";
 
     @Override
-    public void configure() {
-      setName("AnotherWordCountMapReduce");
-      setDescription("WordCount job from Hadoop examples");
+    public void initialize(WorkflowContext context) throws Exception {
+      doneDir = context.getRuntimeArguments().get("done.directory");
```

Comment:
If you have the argument as `done.file`, then you don't even need the `SYNCH_ON_FILE` constant.